backend/src/org/commcare/session/SessionFrame.java

```diff
@@ -95,12 +98,10 @@ public String getFrameId() {
      * This snapshot can be referenced later to compare the eventual state
      * of the frame to an earlier point
      */
-    public void captureSnapshot() {
-        synchronized (steps) {
```

Comment:
when you synchronize at the method level it uses a "hidden" class instance lock.